File: trainee_task/web/src/ts/services/IDepartmentService.ts
import {IDepartment} from "../models/IDepartment";
import {IService} from "./IService";
import jqXHR = JQuery.jqXHR;

export interface IDepartmentService extends IService {

    getAll(): jqXHR;
    getByName(name: string): jqXHR;
    createDepartment(department: IDepartment): jqXHR;
}

File: trainee_task/api/src/main/java/com/aimprosoft/services/IDepartmentService.java
package com.aimprosoft.services;

import com.aimprosoft.exceptions.CRUDException;
import com.aimprosoft.exceptions.ValidateException;
import com.aimprosoft.models.Department;

import java.util.List;

public interface IDepartmentService extends IService<Department> {

    Department createOrUpdate(Department department) throws ValidateException, CRUDException;
    Department getById(Integer id) throws CRUDException;
    Department getDepartmentByName(String departmentName) throws CRUDException;
    List<Department> getAll() throws CRUDException;
}
